lib/posix/pthread.c

```diff
@@ -597,3 +597,46 @@ int pthread_attr_destroy(pthread_attr_t *attr)
 
 	return EINVAL;
 }
+
+/**
+ * @brief Set thread name. Non-portable, extension function
+ *
+ */
+int pthread_setname_np(pthread_t thread, const char *name)
+{
+#ifdef CONFIG_THREAD_NAME
+	k_tid_t kthread = (k_tid_t)thread;
+
+	if ((kthread == NULL)) {
+		return ESRCH;
```

Comment:
"man pthread_setname_np" doesn't explicitly mention ESRCH as a return code. Based on the source code, Glibc would return ENOENT in this case: https://code.woboq.org/userspace/glibc/sysdeps/unix/sysv/linux/pthread_setname.c.html